browser/test/Services/AutoClosingPairsTests.ts

```diff
@@ -1,22 +1,97 @@
 import * as assert from "assert"
 
-import * as AutoClosingPairs from "./../../src/Services/AutoClosingPairs"
+import * as types from "vscode-languageserver-types"
+
+import * as Mocks from "./../Mocks"
+
+import {
+    checkOpenCharacter,
+    getWhiteSpacePrefix,
+    IAutoClosingPair,
+} from "./../../src/Services/AutoClosingPairs"
 
 describe("AutoClosingPairs", () => {
     describe("getWhitespacePrefix", () => {
         it("returns empty if the string doesn't have whitespace", () => {
-            const result = AutoClosingPairs.getWhiteSpacePrefix("test")
+            const result = getWhiteSpacePrefix("test")
             assert.strictEqual(result, "")
         })
 
         it("returns tab", () => {
-            const result = AutoClosingPairs.getWhiteSpacePrefix("\ttest")
+            const result = getWhiteSpacePrefix("\ttest")
             assert.strictEqual(result, "\t")
         })
 
         it("returns spaces", () => {
-            const result = AutoClosingPairs.getWhiteSpacePrefix("  test")
+            const result = getWhiteSpacePrefix("  test")
             assert.strictEqual(result, "  ")
         })
     })
+
+    describe("handleOpenCharacter", () => {
+        let mockEditor: Mocks.MockEditor
+        let mockBuffer: Mocks.MockBuffer
+
+        beforeEach(() => {
+            mockEditor = new Mocks.MockEditor()
+            mockBuffer = new Mocks.MockBuffer("typescript", "test.ts", [""])
+            mockEditor.simulateBufferEnter(mockBuffer)
+        })
+
+        it("inserts both characters of the pair", async () => {
```

Comment:
Great set of test cases! Thanks for going the extra mile and implementing these 💯 Excellent foundation for us as we fix additional bugs or add more functionality here.